src/hotspot/share/oops/oop.inline.hpp

```diff
@@ -392,19 +392,63 @@ oop oopDesc::forward_to_atomic(oop p, atomic_memory_order order) {
   return forwardee();
 }
 
+oop oopDesc::forward_to_self_atomic(markOop compare, atomic_memory_order order) {
+#ifdef _LP64
+  assert(UseAltGCForwarding, "sanity");
```

Comment:
can this assert be moved before #ifdef _LP64?

Reply:
updated